src/main/java/com/kodilla/ecommercee/controller/order/OrderMapper.java

```diff
@@ -22,15 +22,27 @@
     @Autowired
     CartEntityDao cartDao;
 
+    @Autowired
+    OrderDao orderDao;
+
     public OrderEntity mapToOrder(final OrderDto orderDto) throws UserNotFoundException {
-        OrderEntity orderEntity = new OrderEntity(
-                orderDto.getAddress()
-        );
-        orderEntity.setPaid(orderDto.getIsPaid());
-        orderEntity.setUser(userDao.findById(orderDto.getUserId()).orElseThrow(UserNotFoundException::new));
-        orderEntity.setCart(cartDao.findById(orderDto.getCartId()).orElse(null));
 
-        return orderEntity;
+        if(orderDao.findById(orderDto.getId()).isPresent()) {
```

Comment:
Teraz przy wywoływaniu createNewOrder trzeba podać id, bo w przeciwnym wypadku wyskakuje` java.lang.IllegalArgumentException: The given id must not be null!`
Proponuje         `if(orderDto.getId() != null && orderDao.findById(orderDto.getId()).isPresent()) {` ale może wymyślisz coś lepszego :P

Reply:
dodałem tak jak pisałeś. Myślałem by zostawić i zawsze podawać id ale w sumie dla nowych zamówień to bez sensu.